akka-http2-support/src/test/scala/akka/http/impl/engine/http2/ConnectionSpec.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.impl.engine.http2
+
+import scala.concurrent.duration._
+import akka.http.impl.util.{ AkkaSpecWithMaterializer, ExampleHttpContexts }
+import akka.http.scaladsl.{ Http, HttpConnectionContext }
+import akka.http.scaladsl.client.RequestBuilding.Get
+import akka.http.scaladsl.model.HttpResponse
+import akka.http.scaladsl.model.StatusCodes.OK
+import akka.http.scaladsl.server.Directives.complete
+import akka.http.scaladsl.settings.ClientConnectionSettings
+import akka.stream.OverflowStrategy
+import akka.stream.scaladsl.{ Keep, Sink, Source }
+import akka.util.ByteString
+import org.scalatest.concurrent.ScalaFutures
+
+class ConnectionSpec extends AkkaSpecWithMaterializer(
+  """akka.http.server.preview.enable-http2 = on
+    |""".stripMargin) with ScalaFutures {
+  val handler = complete(OK, "Test successful")
+
+  "The HTTP/2 connection API" should {
+    "establish a secure connection" in {
+      val binding =
```

Comment:
Maybe, DRY up those two tests which test the same thing under different circumstances?